src/services/pcn-nat/src/Nat.cpp

```diff
@@ -150,7 +151,23 @@ void Nat::replaceRule(const RuleJsonObject &conf) {
 }
 
 void Nat::delRule() {
-  rule_ = nullptr;
+  rule_ = nullptr; //In this way we can call the destroyers of the rules Snat, Dnat, Port-Forwarding
+  // We need an "empty" rule_ like the one created at initialization in the constructor. 
+  // Otherwise there is an error in the display of the nat after the deletion. As in issue #352
+  RuleJsonObject empty_rule;
+  rule_ = std::make_shared<Rule>(*this, empty_rule); 
+  
+  // TODO: is really necessary?
+  // It seems to be the most logical choice, following the same logic used in the constructor of the Nat. 
+  // The NAT needs to know the IP address of the parent in order to perform the masquerate rules
+  ParameterEventCallback cb = [&](const std::string &parameter, const std::string &value) {
+    external_ip_ = utils::get_ip_from_string(value);
+    logger()->debug("parent IP has been updated to {}", external_ip_);
+    if (rule_->getMasquerade()->getEnabled()) {
+      rule_->getMasquerade()->inject(utils::ip_string_to_nbo_uint(external_ip_));
+    }
+  };
+  subscribe_parent_parameter("ip", cb);
```

Comment:
Good job @pinoOgni.
I don't think this part is needed, the subscribe in the constructor should still be valid, can you test it?

Reply:
Hi @FedeParola thanks for the reply. 

Yesterday I did some tests with and without the subscribe and it doesn't seem to change anything and everything works. I just wanted a second point of view, at this point I remove the subscribe.